model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py

```diff
@@ -351,6 +358,7 @@ def __init__(
         self, exchange: decomposition.ExchangeRuntime = decomposition.SingleNodeExchange()
     ):
         self._exchange = exchange
+        self.halo_exchange_wait = decomposition.create_halo_exchange_wait(self._exchange)
```

Comment:
Can you make this private? 
```suggestion
        self._halo_exchange_wait = decomposition.create_halo_exchange_wait(exchange)
```
and is the `self._exchange`still needed? Otherwise please delete it?

Reply:
Or put differently, I just realised that I don't understand how those 2 the `ExchangeRuntime` and the `ExchangeAndWait` should be used in the `Diffusion` most of the halo exchange calls are as before via the runtime, you only use the ExchangeAndWait at the end, for the single `exchange.wait`. Shouldn't one replace the other or only use it internally?